optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/tri/TriConstraintStream.java

```diff
@@ -50,14 +55,147 @@
      * Exhaustively test each tuple of facts against the {@link TriPredicate}
      * and match if {@link TriPredicate#test(Object, Object, Object)} returns true.
      * <p>
-     * Important: This is slower and less scalable than {@link BiConstraintStream#join(UniConstraintStream, TriJoiner[])}
+     * Important: This is slower and less scalable than {@link BiConstraintStream#join(Class, TriJoiner[])}
```

Comment:
I am ok with doing this change from stream to class,
but all javadoc constraint must be consistent in all Javadocs for constraintstreams.
BiStream's version doesn't change this into Class as far as I can tell.